src/menu/rendersettingsmenustate.cpp

```diff
@@ -503,6 +515,16 @@ std::unique_ptr<MenuState>
     world.getEngine().getEngineConfig()->renderSettings.uiScaleMultiplier = m_uiScaleSelector->getSelectedValue();
     world.getEngine().applySettings();
   }
+  else if(std::get<2>(listBox->getSelected())->getContent() == m_dustDensitySelector)
+  {
+    if(world.getPresenter().getInputHandler().hasDebouncedAction(hid::Action::Left))
+      m_dustDensitySelector->selectPrev();
+    else if(world.getPresenter().getInputHandler().hasDebouncedAction(hid::Action::Right))
+      m_dustDensitySelector->selectNext();
+    world.getEngine().getEngineConfig()->renderSettings.dustDensity
+      = m_dustDensitySelector->getSelectedValue();
+    world.getEngine().applySettings();
+  }
```

Comment:
This pattern repeats over and over again now. Once this PR is done, this should be refactored to use a common non-template base class for `ValueSelector<T>` which would allow for a `dynamic_cast` of `std::get<2>(listBox->getSelected())->getContent()` and calling `selectPrev` and `selectNext` on it, and also use a callback within `ValueSelector<T>` to bind the external value to the widget (in this case, `world.getEngine().getEngineConfig()->renderSettings.dustDensity`).

That's not part of this PR, though.